test/integration/acceptance/tx_heavy_data.cpp

```diff
@@ -0,0 +1,220 @@
+#include "framework/integration_framework/integration_test_framework.hpp"
+#include "interfaces/utils/specified_visitor.hpp"
+#include "module/shared_model/builders/protobuf/test_query_builder.hpp"
+#include "module/shared_model/builders/protobuf/test_transaction_builder.hpp"
+#include "validators/permissions.hpp"
+
+using namespace std::string_literals;
+using namespace integration_framework;
+using namespace shared_model;
+
+class HeavyTransactionTest : public ::testing::Test {
+ public:
+  /**
+   * Creates the transaction with the user creation commands
+   * @param perms are the permissions of the user
+   * @return built tx and a hash of its payload
+   */
+  auto makeUserWithPerms(
+      const std::vector<std::string> &perms = {
+          shared_model::permissions::role_perm_group.begin(),
+          shared_model::permissions::role_perm_group.end()}) {
+    return framework::createUserWithPerms(
+               kUser, kUserKeypair.publicKey(), "role"s, perms)
+        .build()
+        .signAndAddSignature(kAdminKeypair);
+  }
+
+  /**
+   * Create valid base pre-built transaction
+   * @return pre-built tx
+   */
+  auto baseTx() {
+    return TestUnsignedTransactionBuilder()
+        .txCounter(1)
+        .creatorAccountId(kUserId)
+        .createdTime(iroha::time::now());
+  }
+
+  /**
+   * Generate stub of transaction for setting data to default user account
+   * @param key - key for the insertion
+   * @param value - data that will be attached
+   * @return generated builder, without signature
+   */
+  auto setAcountDetailTx(const std::string &key, const std::string &value) {
+    return baseTx().setAccountDetail(kUserId, key, value);
+  }
+
+  /**
+   * Util method for stub data generation
+   * @param quantity - number of bytes
+   * @return new string with passed quantity length
+   */
+  static auto generateData(size_t quantity) {
+    return std::string(quantity, 'F');
+  }
+
+  /**
+   * Sign pre-built object
+   * @param builder is a pre-built signable object
+   * @return completed object
+   */
+  template <typename Builder>
+  auto complete(Builder builder) {
+    return builder.build().signAndAddSignature(kUserKeypair);
+  }
+
+  /**
+   * Create valid basis of pre-built query
+   * @return query stub with counter, creator and time
+   */
+  auto baseQuery() {
+    return TestUnsignedQueryBuilder()
+        .queryCounter(1)
+        .creatorAccountId(kUserId)
+        .createdTime(iroha::time::now());
+  }
+
+  const std::string kUser = "user"s;
+  const std::string kUserId =
+      kUser + "@"s + IntegrationTestFramework::kDefaultDomain;
+  const crypto::Keypair kUserKeypair =
+      crypto::DefaultCryptoAlgorithmType::generateKeypair();
+  const crypto::Keypair kAdminKeypair =
+      crypto::DefaultCryptoAlgorithmType::generateKeypair();
+};
+
+/**
+ * @given some user with all required permissions
+ * @when send tx with addAccountDetail with big, but stateless valid data inside
+ * @then transaction is passed
```

Comment:
Should say that transaction is not passed due to big data inside, right?

Reply:
Yes, I'll fix a description. "but stateless **in**valid data"